setup.py

```diff
@@ -53,6 +56,7 @@ def get_build_triple(build_mpi):
     os.environ['ARCHFLAGS'] = f'-arch {machine}'
     libcharm_filename = 'libcharm.dylib'
     charmrun_filename = 'charmrun'
+    os.environ['CPPFLAGS'] = f'-Wno-error=implicit-function-declaration' #needed because this error will cause builds to fail if the Clang compiler is version 15.0.0 or newer (it is only a warning on older builds)
```

Comment:
This seems to replace all CPPFLAGS, should it maybe just append this flag?

Reply:
any attempt to append on my end throws a KeyError. i tried an if statement to check if the field exists, but it throws the keyerror on that too. any ideas?

Reply:
never mind, i think i found a way. is the new version a good idea?

Reply:
done